Library/Homebrew/dev-cmd/determine-cask-runners.rb

```diff
@@ -0,0 +1,351 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "abstract_command"
+require "tap"
+require "utils/github/api"
+require "cli/parser"
+require "system_command"
+
+module Homebrew
+  module DevCmd
+    class DetermineCaskRunners < AbstractCommand
+      MAX_JOBS = 256
+
+      # Weight for each arch must add up to 1.0.
+      INTEL_RUNNERS = T.let({
+        { symbol: :monterey, name: "macos-12", arch: :intel } => 0.0,
+        { symbol: :ventura,  name: "macos-13", arch: :intel } => 1.0,
+      }.freeze, T::Hash[T.untyped, T.untyped])
```

Comment:
Can we get rid of the `T.untyped`s in this file?

Reply:
I have given this a go, I'm brand new to typing in Ruby, so I'm not sure if my solutions are correct.